File: app/src/main/java/mx/qsistemas/infracciones/Utils/MD5.java
package mx.qsistemas.infracciones.Utils;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

/**
 * Developed by ingmtz on 10/21/16.
 */

public class MD5 {

    public static String toMD5(String s) {
        try {
            MessageDigest digest = java.security.MessageDigest.getInstance("MD5");
            digest.update(s.getBytes());
            byte messageDigest[] = digest.digest();

            StringBuilder hexString = new StringBuilder();

            for (byte msg : messageDigest) {
                String h = Integer.toHexString(0xFF & msg);
                while (h.length() < 2)
                    h = "0" + h;
                hexString.append(h);
            }

            return hexString.toString();

        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }

        return "";
    }

}

File: app/src/main/java/mx/qsistemas/infracciones/utils/MD5.kt
package mx.qsistemas.infracciones.utils

import java.security.NoSuchAlgorithmException

class MD5 {
    companion object {
        fun toMD5(s: String): String {
            try {
                val digest = java.security.MessageDigest.getInstance("MD5")
                digest.update(s.toByteArray())
                val messageDigest = digest.digest()
                val hexString = StringBuilder()
                for (msg in messageDigest) {
                    var h = Integer.toHexString(0xFF and msg.toInt())
                    while (h.length < 2)
                        h = "0$h"
                    hexString.append(h)
                }
                return hexString.toString().toUpperCase()
            } catch (e: NoSuchAlgorithmException) {
                e.printStackTrace()
            }
            return ""
        }
    }
}